maro/cli/grass/lib/services/master_agent/agent.py

```diff
@@ -124,13 +124,24 @@ def update_name_to_job_details(self) -> None:
         # Save jobs details.
         for job_name, job_details in name_to_job_details.items():
             job_details["check_time"] = self._redis_controller.get_time()
-            self._redis_controller.set_job_details(
-                job_name=job_name,
-                job_details=job_details
-            )
+            if job_details["containers"] != {}:
+                for container_name, container_details in job_details["containers"].items():
+                    if container_details["state"]["Status"] == "running":
+                        job_state = JobStatus.RUNNING
+                        break
```

Comment:
“break” is not needed. Please remove all of them in this code block.

Reply:
Each job may start many containers, use break to avoid part of containers finish, but others still running. Same reason for failed.